### DISCUSSION TITLE
vscode markown auto-completions/auto-suggestions

### DISCUSSION THREAD

**Discussion by psadi** [marp-team/marp#417] [Ideas] [answered]
It would be nice to have autocompletions, suggestions with available options while creating slides in vscode

thoughts ?

**Comment by yhatt** [comment, marked as answer]
Marp for VS Code has already supported auto completion for directives. Aren't you missing out the documentation?
https://github.com/marp-team/marp-vscode#auto-completion

[![](https://raw.githubusercontent.com/marp-team/marp-vscode/main/docs/auto-completion.gif)](https://github.com/marp-team/marp-vscode#auto-completion)

**Reply by niraj8** [reply]
Is there any restriction on the suggestion window opening automatically when Intellisense suggestions are available for the field? 

Instead of pressing the `Ctrl + Space` or `Opt + Esc`, it would be great if the dropdown appears automatically.

**Reply by yhatt** [reply]
Auto completion of directives can use in Markdown front-matter and HTML comments. Especially HTML comment allows either of directives definition or presenter notes.

When writing presenter notes, flickering autocomplete popup and trapping keyboard controls are annoying. So we have decided to require manual operation for opening auto completion. It would be a difficult task to detect automatically if should open auto completion in a position that every user think comfortable.

**Comment by psadi** [comment]
Yes, For me hitting `ctrl + space` pops the suggestion. I didnt try this before.

This is sufficient for my problem statement. I was hoping for auto suggestions like to know all the possible `key: values` without the needs to scraping the docs.
